src/app/clusters/boolean-state-server/boolean-state-cluster.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/EventLogging.h>
+#include <app/clusters/boolean-state-server/boolean-state-cluster.h>
+#include <app/server-cluster/AttributeListBuilder.h>
+#include <clusters/BooleanState/Metadata.h>
+
+namespace chip::app::Clusters {
+
+using namespace BooleanState::Attributes;
+
+BooleanStateCluster::BooleanStateCluster(EndpointId endpointId) :
+    DefaultServerCluster({ endpointId, BooleanState::Id }), mStateValue(false)
+{}
+
+DataModel::ActionReturnStatus BooleanStateCluster::ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                                 AttributeValueEncoder & encoder)
+{
+    switch (request.path.mAttributeId)
+    {
+    case StateValue::Id:
+        return encoder.Encode(mStateValue);
+    case ClusterRevision::Id:
+        return encoder.Encode(BooleanState::kRevision);
+    case FeatureMap::Id:
+        return encoder.Encode<uint32_t>(0);
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedAttribute;
+    }
+}
+
+CHIP_ERROR BooleanStateCluster::Attributes(const ConcreteClusterPath & path,
+                                           ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder)
+{
+    AttributeListBuilder listBuilder(builder);
+    return listBuilder.Append(Span(BooleanState::Attributes::kMandatoryMetadata), {});
+}
+
+CHIP_ERROR BooleanStateCluster::SetStateValue(const StateValue::TypeInfo::Type & stateValue, EventNumber & eventNumber)
+{
+    VerifyOrReturnError(mStateValue != stateValue, CHIP_NO_ERROR);
+    mStateValue = stateValue;
+    NotifyAttributeChanged(StateValue::Id);
+    BooleanState::Events::StateChange::Type event{ stateValue };
+    return app::LogEvent(event, mPath.mEndpointId, eventNumber);
```

Comment:
You should use the context to log events, not globals (so that things are unit testable):

https://github.com/project-chip/connectedhomeip/blob/master/src/app/data-model-provider/Context.h#L34

https://github.com/project-chip/connectedhomeip/blob/master/src/app/data-model-provider/EventsGenerator.h#L88

You do not need to propagate the error as we generally can just "log"